compiler/rustc_index/src/vec.rs

```diff
@@ -12,10 +12,12 @@ use std::vec;
 use crate::{Idx, IndexSlice};
 
 /// An owned contiguous collection of `T`s, indexed by `I` rather than by `usize`.
+/// Its purpose is to avoid mixing indexes.
 ///
 /// While it's possible to use `u32` or `usize` directly for `I`,
 /// you almost certainly want to use a [`newtype_index!`]-generated type instead.
 ///
+/// This allows to index the IndexVec with the new index type.
```

Comment:
I'm not entirely sure but I think you might need an empty line after this one to make the `[newtype_index!]` link resolve. 

```suggestion
/// This allows to index the IndexVec with the new index type.
///
```